src/renderers/common/Renderer.js

```diff
@@ -147,7 +147,12 @@ class Renderer {
 
 		this.debug = {
 			checkShaderErrors: true,
-			onShaderError: null
+			onShaderError: null,
+			getRawShader: ( object ) => {
```

Comment:
You suggested in https://github.com/mrdoob/three.js/issues/28984#issuecomment-2253318912 scene and camera as additional parameters but I'm not sure what to do with them for recreating the original logic of the example. It seems just the 3D object is sufficient.

Reply:
The problem with sending just the object is that it will create a shader from scratch, ignoring the `RenderContext`, which includes `RenderTarget`, type of camera, scene.environment, etc. The object may have shader variation if it is inside a `RenderBundle` currently `group.static=true`, for this reason we will need to compile the entire scene as well, and then get the real shader code.

Reply:
I'll see what I can do to help here by editing the PR.